### ISSUE TITLE
test(api): coverage

### ISSUE THREAD

**Issue by slekup** [odnlabs/opendevnet#275]
Add coverage for API testing then upload artifacts in GitHub CI workflow.

**Comment by A-alharbi9** [comment]
I will work on this issue, @slekup.

**Comment by slekup** [comment]
Awesome! I'm not sure how it's done in rust, so will be interesting. Do you have write access to create a new branch?

**Comment by A-alharbi9** [comment]
Yes, I have write access. It's quite interesting, because I'm still new to Rust. I'll keep you updated about it and close this issue once I'm done.

**Comment by slekup** [comment]
I'm still pretty new to rust as well, but I'm loving it so far. I'm developing a [CLI tool](https://github.com/slekup/blue), which I will add to this project once the important features are developed.
It will replace husky, lint-staged, commitlint, and most of the bash scripts. With the tool, backend developers won't have to install any frontend dependencies to setup the environment, and vise-versa. There are a lot of other features I have in mind, but these are most relevant to ODN right now.

By the way, just some information about the pr that I haven't added to the internal docs yet. In the pr:
- Use the same convention for commits for the title.
- Reference the issue in the description.
- The pr will be squashed into the dev branch.
- If the pr changes are too large to be squashed, create sub-branches and squash those into the task branch, and I'll rebase the task branch into dev.
